server/drivers/hd44780-4bit.c

```diff
@@ -262,9 +268,40 @@ lcdstat_HD44780_senddata(PrivateData *p, unsigned char displayID, unsigned char
  */
 void lcdstat_HD44780_backlight(PrivateData *p, unsigned char state)
 {
+	static unsigned char old_state=0;
+	unsigned char brightnessLevelVFD = 0;
+
 	p->backlight_bit = ((!p->have_backlight||state)?0:BL);
 
 	port_out(p->port, p->backlight_bit);
+
+	if (p->isVFDDisplay)
+	{
+		if (state!=old_state)
+		{
```

Comment:
No need for nested if statements.
`if (p->isVFDDisplay && state != old_state)` is easier to read I think.